mne/coreg.py

```diff
@@ -1223,3 +1227,576 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+            weights.append(np.full(len(mri_pts[-1]), self._hpi_weight))
+        head_pts = np.concatenate(head_pts)
+        mri_pts = np.concatenate(mri_pts)
+        weights = np.concatenate(weights)
+        if self._n_scale_param == 0:
+            mri_pts *= self._parameters[6:9]  # not done in fit_matched_points
+        return head_pts, mri_pts, weights
+
+    def fit_icp(self, n_iterations=20, lpa_weight=1., nasion_weight=10.,
+                rpa_weight=1.):
+        """Find MRI scaling, translation, and rotation to match HSP.
+
+        Parameters
+        ----------
+        n_iterations : int
+            Maximum number of iterations.
+        lpa_weight : float
+            Relative weight for LPA. The default value is 1.
+        nasion_weight : float
+            Relative weight for nasion. The default value is 10.
+        rpa_weight : float
+            Relative weight for RPA. The default value is 1.
+        """
+        self._lpa_weight = lpa_weight
+        self._nasion_weight = nasion_weight
+        self._rpa_weight = rpa_weight
+
+        # Initial guess (current state)
+        est = self._parameters[:[6, 7, None, 9][self._n_scale_param]]
+
+        # Do the fits, assigning and evaluating at each step
+        for _ in range(n_iterations):
+            head_pts, mri_pts, weights = self._setup_icp()
+            est = fit_matched_points(mri_pts, head_pts,
+                                     scale=self._n_scale_param,
+                                     x0=est, out='params', weights=weights)
+            if self._n_scale_param == 0:
+                tmp_est = self._parameters.copy()
+                tmp_est[:6] = est
+                self._update_params(tmp_est)
+            elif self._n_scale_param == 1:
+                self._update_params(list(est) + [est[-1]] * 2)
+            else:
+                self._update_params(est)
+            angle, move, scale = self._changes
+            if angle <= self._icp_angle and move <= self._icp_distance and \
+                    all(scale <= self._icp_scale):
+                break
+
+    def omit_hsp_points(self, distance):
+        """Exclude head shape points that are far away from the MRI head.
+
+        Parameters
+        ----------
+        distance : float
+            Exclude all points that are further away from the MRI head than
+            this distance. A value of distance <= 0 excludes nothing.
+        """
+        distance = float(distance)
+        if distance <= 0:
+            return
+
+        # find the new filter
+        mask = self._orig_hsp_point_distance <= distance
+        n_excluded = np.sum(~mask)
+        logger.info("Coregistration: Excluding %i head shape points with "
+                    "distance >= %.3f m.", n_excluded, distance)
+        # set the filter
+        with warnings.catch_warnings(record=True):  # comp to None in Traits
+            self._dig.extra_points_filter = mask
+
+    def point_distance(self):
```

Comment:
can you we have a more explicit name? it's all dig points and scalp surface? also I would use plural to explicit it's all distances. eg


```suggestion
    def compute_dig_head_distances(self):
```

?